pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1339,7 +1339,32 @@ func (c *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID strin
 
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (c *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
-	return false, cloudprovider.NotImplemented
+	instanceID, err := kubernetesInstanceID(providerID).mapToAWSInstanceID()
+	if err != nil {
+		return false, err
+	}
+
+	request := &ec2.DescribeInstancesInput{
+		InstanceIds: []*string{instanceID.awsString()},
+	}
+
+	instances, err := c.ec2.DescribeInstances(request)
+	if err != nil {
+		return false, err
+	}
+	if len(instances) == 0 {
+		return false, nil
+	}
+	if len(instances) > 1 {
+		return false, fmt.Errorf("multiple instances found for instance: %s", instanceID)
+	}
+
+	state := instances[0].State.Name
+	// valid state for detaching volumes is stopped 
+	if *state == "stopped" {
```

Comment:
I think also `terminated`.

Might also be better to use the constants in the ec2 sdk: `InstanceStateNameStopped` and `InstanceStateNameTerminated`

Reply:
@justinsb can you read my longer comment above? The thing here is that with nodelifecycle controller this thing is going to work pretty well. Nodes are not deleted from cluster if those are in some weird state like pending. However, in future with cloud controller manager this thing is not going to work. We have two interfaces where we should check all possible states of instance. Otherwise those are deleted from cluster. Currently we check only stopped, terminated(? i will add it) and active. We should add states `shutting-down` `stopping` `pending` and `rebooting` instance states to InstanceShutdownByProviderID or InstanceExistsByProviderID. Imo exists is correct place for these.